python/lsst/jointcal/jointcal.py

```diff
@@ -419,32 +503,260 @@ class JointcalTask(pipeBase.PipelineTask, pipeBase.CmdLineTask):
         else:
             self.photometryRefObjLoader = None
 
         # To hold various computed metrics for use by tests
         self.job = Job.load_metrics_package(subset='jointcal')
 
+    def runQuantum(self, butlerQC, inputRefs, outputRefs):
+        # We override runQuantum to set up the refObjLoaders and write the
+        # outputs to the correct refs.
+        inputs = butlerQC.get(inputRefs)
+        # We want the tract number for writing debug files
+        tract = butlerQC.quantum.dataId['tract']
+        if self.config.doAstrometry:
+            self.astrometryRefObjLoader = ReferenceObjectLoader(
+                dataIds=[ref.datasetRef.dataId
+                         for ref in inputRefs.astrometryRefCat],
+                refCats=inputs.pop('astrometryRefCat'),
+                config=self.config.astrometryRefObjLoader,
+                log=self.log)
+        if self.config.doPhotometry:
+            self.photometryRefObjLoader = ReferenceObjectLoader(
+                dataIds=[ref.datasetRef.dataId
+                         for ref in inputRefs.photometryRefCat],
+                refCats=inputs.pop('photometryRefCat'),
+                config=self.config.photometryRefObjLoader,
+                log=self.log)
+        outputs = self.run(**inputs, tract=tract)
+        if self.config.doAstrometry:
+            self._put_output(butlerQC, outputs.outputWcs, outputRefs.outputWcs)
+        if self.config.doPhotometry:
+            self._put_output(butlerQC, outputs.outputPhotoCalib, outputRefs.outputPhotoCalib)
+
+    def _put_output(self, butlerQC, outputs, outputRefs):
+        """Butler put the output datasets to their appropriate datarefs.
+
+        Parameters
+        ----------
+        butlerQC : `ButlerQuantumContext`
+            A butler which is specialized to operate in the context of a
+            `lsst.daf.butler.Quantum`.
+        outputs : `dict` [`tuple`, `lsst.afw.geom.SkyWcs`] or
+                  `dict` [`tuple, `lsst.afw.image.PhotoCalib`]
+            The fitted objects to put.
+        outputRefs : `list` [`OutputQuantizedConnection`]
+            The DatasetRefs to put the data to.
+        """
+        for ref in outputRefs:
+            key = (ref.dataId['visit'], ref.dataId['detector'])
+            butlerQC.put(outputs[key], ref)
+
+    def run(self, inputSourceTableVisit, inputVisitSummary, inputCamera, tract=None):
+        # Docstring inherited.
+        sourceFluxField = "slot_%sFlux" % (self.config.sourceFluxType,)
+        jointcalControl = lsst.jointcal.JointcalControl(sourceFluxField)
+        associations = lsst.jointcal.Associations()
+        self.focalPlaneBBox = inputCamera.getFpBBox()
+        oldWcsList, bands = self._load_data(inputSourceTableVisit,
+                                            inputVisitSummary,
+                                            associations,
+                                            jointcalControl)
+
```

Comment:
Could you add a code comment here, describing briefly what this next step is doing?

Reply:
Sorry, which step are you referring to here?

Reply:
Sorry, this comment should have been one line lower: `boundingCircle, center, radius, defaultFilter = self._prep_sky(associations, bands)`

Reply:
That seems like it would be duplicating the docstring on `_prep_sky`, no?

Reply:
Yes. If that seems excessive to you, feel free to skip this.